pkg/formatter/txt.go

```diff
@@ -96,29 +72,23 @@ func StoreCookbooksReportTXT(records []*reporting.CookbookRecord) error {
 
 		// verify errors
 		// TODO @afiune we could refactor this to not be duplicated on every report
+		// TODO @marcparadise this might be a side effect of the original way we
+		//                    has of processing each phase separately; it might be we
+		//                    can stop differentiating between the type of error
+		//                    in the  CookbookRecord, and just have an error array in the struct.
 		if record.DownloadError != nil {
-			anyError = true
-			downloadErrors.WriteString(
+			errorBuilder.WriteString(
 				fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.DownloadError))
 		}
 		if record.CookstyleError != nil {
-			anyError = true
-			cookstyleErrors.WriteString(
+			errorBuilder.WriteString(
 				fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.CookstyleError))
 		}
 		if record.UsageLookupError != nil {
-			anyError = true
-			usageFetchErrors.WriteString(
+			errorBuilder.WriteString(
 				fmt.Sprintf(" - %s (%s): %v\n", record.Name, record.Version, record.UsageLookupError))
 		}
 
-		reportFile.WriteString(strBuilder.String())
-	}
-
-	fmt.Printf("\nCookbooks report generated at: \n  => %s\n", reportPath)
-	if anyError {
-		return StoreErrorsFromBuilders(timestamp, downloadErrors, cookstyleErrors, usageFetchErrors)
 	}
-
-	return nil
+	return strBuilder.String(), errorBuilder.String()
```

Comment:
I'm thinking about this "Error String" thing. I am not sure Im ok with it since a golang `error` is nothing more than a struct that has to fulfill the error interface: 
```go
type error interface {
    Error() string
}
```
That means that you can return an `error` and still translate it as a string with `err.Error()` func docs [here](https://blog.golang.org/error-handling-and-go). But I haven't thought this through, I will need a min.

Reply:
I was looking into that as well, but I couldn't get it to 'feel' right.  I think of the `error` type as a thing that happens when a failure occurs, at the time it occurs. 

In this case, there's no failure occurring in the generation of the report - so it seems wrong to return an `error` type.

Reply:
@marcparadise @tyler-ball What about a `Formater` struct that contains both strings? 🤔
```go
type Formatter struct {
    Report string
    Error  string
}
```

Reply:
Win! I'll do that.